### PR TITLE
feat(build) feat(ci) Force npm registry for npmjs.com + Use npm ci in travis + Add package-lock.json to repo

### PR DESCRIPTION
**Pull request by SuperITMan**
feat(build) feat(ci) Force npm registry for npmjs.com + Use npm ci in travis + Add package-lock.json to repo

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

This contains package-lock.json file and .npmrc for forcing the the registry url to https://registry.npmjs.com.
It uses also npm ci for Travis build.

It also adds some badges on README page

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[X] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #17 #23 #43  #249  

## What is the new behavior?
Support of npm ci

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other information

### CODE REVIEW

**Comment by SuperITMan** [comment]
I'm still working on it. Please don't merge currently

**Comment by dsebastien** [comment]
Can you integrate my bonus commit?
https://github.com/NationalBankBelgium/stark/commit/e4ad28e74691a0109d611a2b1b2fa331d720b10f

File: packages/stark-core/package.json [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
   "name": "@nationalbankbelgium/stark-core",
   "version": "0.0.0-PLACEHOLDER-VERSION",
-  "module": "./esm5/stark-core.js",
-  "es2015": "./es2015/stark-core.js",
+  "module": "./fesm5/stark-core.js",
+  "es2015": "./fesm2015/stark-core.js",
+  "esm5": "./esm5/stark-core.js",
+  "esm2015": "./esm2015/stark-core.js",
+  "fesm5": "./fesm5/stark-core.js",
+  "fesm2015": "fesm2015/stark-core.js",
```
Comment (christophercr) [inline packages/stark-core/package.json]: Ok, so I think this is the change related to APF v6 right?

One thing, the path of the fesm2015 doesn't have './'like the others

File: packages/stark-testing/rollup.config.js [hunk 2]
```diff
@@ -1,20 +1,16 @@
...
 
-const resolve = require("rollup-plugin-node-resolve");
-const commonjs = require("rollup-plugin-commonjs");
+const commonData = require("../rollup.config.common-data.js"); // common configuration between environments
 
 module.exports = {
-	//export default {
 	input: "../../dist/packages-dist/stark-testing/karma.conf.typescript.js",
+	external: commonData.external,
+	plugins: commonData.plugins,
```
Comment (christophercr) [inline packages/stark-testing/rollup.config.js]: Do we really want to include globals and all the rollup plugins here? Normally the stark-testing package just contains scripts and npm dependencies...

Comment (dsebastien) [reply packages/stark-testing/rollup.config.js]: I went from the assumption that we might later also add utility code in there. But that's not for now indeed and when we get there we'll probably rework that config

File: packages/stark-core/package.json (hunk 1 above)
Comment (dsebastien) [reply packages/stark-core/package.json]: Indeed, APF v6.
And there's a mistake indeed, there should be a ./ in front (damn ^^)

**Pull request merged**